frontend/src/app/main/data/workspace/variants.cljs

```diff
@@ -310,3 +347,38 @@
            (rx/from (map add-new-variant selected-ids))
            (rx/of (dwu/commit-undo-transaction undo-id)))
           (rx/of (dws/duplicate-selected true)))))))
+
+
+(defn rename-variant
+  [variant-id name]
+  (ptk/reify ::rename-variant
+
+    ptk/WatchEvent
+    (watch [_ state _]
+      (let [data               (dsh/lookup-file-data state)
+            objects            (dsh/lookup-page-objects state)
+            variant-components (cfv/find-variant-components data objects variant-id)
```

Comment:
And here you should avoid duplicate lookups using `dsh/get-page` + `(get page :objects)` istead of `(dsh/lookup-page-objects state)`